src/v0/destinations/moengage/transform.js

```
@@ -50,57 +119,16 @@ function responseBuilderSimple(message, category, destination) {
-        // nested attributes are not by moengage so it is falttened
-        payload.attributes = flattenJson(payload.attributes);
-        break;
-      case 'device':
-        // Ref: https://docs.moengage.com/docs/data-import-apis#device-api
-        payload.type = 'device';
-        payload.attributes = constructPayload(
-          message,
-          MAPPING_CONFIG[CONFIG_CATEGORIES.DEVICE_ATTR.name],
-        );
-        // nested attributes are not by moengage so it is falttened
-        payload.attributes = flattenJson(payload.attributes);
-
-        // Ref - https://developers.moengage.com/hc/en-us/articles/4413167466260-Device-
-        if (isAppleFamily(payload.attributes?.platform)) {
-          payload.attributes.platform = 'iOS';
-        }
-        break;
-      case 'track':
-        // Ref: https://docs.moengage.com/docs/data-import-apis#event-api
-        payload.type = 'event';
-        payload.actions = [
-          constructPayload(message, MAPPING_CONFIG[CONFIG_CATEGORIES.TRACK_ATTR.name]),
-        ];
 
-        // Ref - https://developers.moengage.com/hc/en-us/articles/4413174104852-Event-
-        if (isAppleFamily(payload.actions[0]?.platform)) {
-          payload.actions[0].platform = 'iOS';
-        }
-        break;
-      case EventType.ALIAS:
-        // clean as per merge user call in moengage
-        delete response.headers['MOE-APPKEY'];
-        break;
-      default:
-        throw new InstrumentationError(`Event type ${category.type} is not supported`);
-    }
-
-    response.body.JSON = removeUndefinedAndNullValues(payload);
-  } else {
-    // fail-safety for developer error
-    throw new TransformationError('Payload could not be constructed');
+  if (category.type === 'alias') {
+    delete response.headers['MOE-APPKEY'];
+    response.body.JSON = removeUndefinedAndNullValues(basePayload);
+    return response;
+  }
+
+  const payload = createDestinationPayload({ message, category, useObjectData });
+  if (payload) {
```

should we fail if payload is empty ? other-wise, response gets returned with no body

---

Yes we should throw an error when payload is empty